pkg/controller/storageversionmigrator/storageversionmigrator.go

```diff
@@ -235,17 +249,15 @@ func (svmc *SVMController) sync(ctx context.Context, key string) error {
 		return errStatus
 	}
 
-	gcListResourceVersion, err := convertResourceVersionToInt(resourceMonitor.Controller.LastSyncResourceVersion())
+	gcListResourceVersion := resourceMonitor.Controller.LastSyncResourceVersion()
+	listResourceVersion := toBeProcessedSVM.Status.ResourceVersion
+
+	rvCmp, err := resourceversion.CompareResourceVersion(gcListResourceVersion, listResourceVersion)
 	if err != nil {
-		return err
+		return fmt.Errorf("error comparing resource versions between GC and SVM resource: %w", err)
```

Comment:
I think this should fail the migration right?  Retrying will never fix it?

Reply:
Done! Also added test cases for this.